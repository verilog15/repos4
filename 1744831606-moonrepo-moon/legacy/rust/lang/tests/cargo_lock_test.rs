use moon_rust_lang::cargo_lock::*;
use moon_test_utils::{assert_debug_snapshot, create_sandbox};

#[test]
fn resolves_lockfile_dep_checksums() {
    let sandbox = create_sandbox("rust/workspaces");
    let deps = load_lockfile_dependencies(sandbox.path().join("Cargo.lock")).unwrap();

    assert_debug_snapshot!(deps);
}

#[test]
fn resolves_lockfile_dep_checksums_for_nonworkspace() {
    let sandbox = create_sandbox("rust/project");
    let deps = load_lockfile_dependencies(sandbox.path().join("Cargo.lock")).unwrap();

    assert_debug_snapshot!(deps);
}
