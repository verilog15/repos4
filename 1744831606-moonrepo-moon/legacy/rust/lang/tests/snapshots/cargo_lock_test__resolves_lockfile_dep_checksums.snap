---
source: legacy/rust/lang/tests/cargo_lock_test.rs
expression: deps
---
{
    "inherited-dep": [
        "0.0.1",
    ],
    "path-deps": [
        "0.0.1",
    ],
    "serde_derive": [
        "291a097c63d8497e00160b166a967a4a79c64f3facdd01cbd7502231688d77df",
    ],
    "serde": [
        "bb2f3770c8bce3bcda7e149193a069a0f4365bda1fa5cd88e03bca26afc1216c",
    ],
    "syn": [
        "a34fcf3e8b60f57e6a14301a2e916d323af98b0ea63c599441eec8558660c822",
    ],
    "proc-macro2": [
        "2b63bdb0cd06f1f4dedf69b254734f9b45af66e4a031e42a7480257d9898b435",
    ],
    "bin-crate": [
        "0.0.1",
    ],
    "cd_env": [
        "08a91cc18015c41dd5d117c356b1708a23a0469c3256403af09dc2a48a7879ce",
    ],
    "ci_env": [
        "b8ec45c59e470d7c361345776fea051dd4d3e9c2db8ed0d3fb343cbaf0f2b50b",
    ],
    "quote": [
        "4424af4bf778aae2051a77b60283332f386554255d722233d09fbfc7e30da2fc",
    ],
    "normal-dep": [
        "0.0.1",
    ],
    "unicode-ident": [
        "e5464a87b239f13a63a501f2701565754bae92d243d4bb7eb12f6d57d2269bf4",
    ],
}
